conformance/utils/suite/experimental_reports.go

```diff
@@ -0,0 +1,191 @@
+
+type testResult struct {
+	test   ConformanceTest
+	result resultType
+}
+
+type resultType string
+
+var (
+	testSucceeded    resultType = "SUCCEEDED"
+	testFailed       resultType = "FAILED"
+	testSkipped      resultType = "SKIPPED"
+	testNotSupported resultType = "NOT_SUPPORTED"
+)
+
+type profileReportsMap map[ConformanceProfileName]confv1a1.ProfileReport
+
+func newReports() profileReportsMap {
+	return make(profileReportsMap)
+}
+
+func (p profileReportsMap) addTestResults(conformanceProfile ConformanceProfile, result testResult) {
+	// initialize the profile report if not already initialized
+	if _, ok := p[conformanceProfile.Name]; !ok {
+		p[conformanceProfile.Name] = confv1a1.ProfileReport{
+			Name: string(conformanceProfile.Name),
+		}
+	}
+
+	testIsExtended := isTestExtended(conformanceProfile, result.test)
+	report := p[conformanceProfile.Name]
+
+	switch result.result {
+	case testSucceeded:
+		if testIsExtended {
+			if report.Extended == nil {
+				report.Extended = &confv1a1.ExtendedStatus{}
+			}
+			report.Extended.Statistics.Passed++
+
+		} else {
+			report.Core.Statistics.Passed++
+		}
+	case testFailed:
+		if testIsExtended {
+			if report.Extended == nil {
+				report.Extended = &confv1a1.ExtendedStatus{}
+			}
+			if report.Extended.FailedTests == nil {
+				report.Extended.FailedTests = []string{}
+			}
+			report.Extended.FailedTests = append(report.Extended.FailedTests, result.test.ShortName)
+			report.Extended.Statistics.Failed++
+		} else {
+			report.Core.Statistics.Failed++
+			if report.Core.FailedTests == nil {
+				report.Core.FailedTests = []string{}
+			}
+			report.Core.FailedTests = append(report.Core.FailedTests, result.test.ShortName)
+		}
+	case testSkipped:
+		if testIsExtended {
+			if report.Extended == nil {
+				report.Extended = &confv1a1.ExtendedStatus{}
+			}
+			report.Extended.Statistics.Skipped++
+			if report.Extended.SkippedTests == nil {
+				report.Extended.SkippedTests = []string{}
+			}
+			report.Extended.SkippedTests = append(report.Extended.SkippedTests, result.test.ShortName)
+		} else {
+			report.Core.Statistics.Skipped++
+			if report.Core.SkippedTests == nil {
+				report.Core.SkippedTests = []string{}
+			}
+			report.Core.SkippedTests = append(report.Core.SkippedTests, result.test.ShortName)
+		}
+	}
+	p[conformanceProfile.Name] = report
+}
+
+func (p profileReportsMap) list() (profileReports []confv1a1.ProfileReport) {
+	for _, profileReport := range p {
+		profileReports = append(profileReports, profileReport)
+	}
+	return
+}
+
+func (p profileReportsMap) compileResults(supportedFeaturesMap map[ConformanceProfileName]sets.Set[SupportedFeature], unsupportedFeaturesMap map[ConformanceProfileName]sets.Set[SupportedFeature]) {
+	for key, report := range p {
+		// report the overall result for core features
+		if report.Core.Failed > 0 {
+			report.Core.Result = confv1a1.Failure
+		} else if report.Core.Skipped > 0 {
```

Comment:
shouldnt the existence of unsupported tests be also treated as a `partial` result ?

Reply:
We can see the extended features badge in two different ways:
1. Only implementations supporting ALL the extended features can have the `success` badge on them: implementations supporting only a subset of features with no failing test, get a `partial` badge
2. implementations supporting some extended features (even if not all of them) can have the `success` badge, along with the list of `supported/not-supported` features. In this scenario, no `partial` badge can be given, as either all the tests for a subset of tests pass, and the `success` badge is given, or some test fails, and the badge is `failure`.

In this implementation, we pursued the second way. I personally lean toward this implementation, as for extended tests we can select the specific features we are compliant with, so that an implementation can claim to be core conformant, and conformant with `x,y,z` extended features, and receive an according `success` badge.  I'm curious to hear your thoughts,  I think there is room for discussion here.

@shaneutt, any thoughts on this aspect?

Reply:
so the issue here is reg readability, as a user I need to interpret `Result`  + `UnsupportedFeatures` to understand the true conformance level, Im proposing a tiered approach (`Result` encapsulates `UnsupportedFeatures`) so the user first interprets the `Result` and then looks at other fields to understand the WHY

Reply:
Here is the output for the extended section:
```json
"extended": {
        "status": {
          "result": "success",
          "summary": "",
          "statistics": {
            "Passed": 1,
            "Skipped": 0,
            "Failed": 0
          }
        },
        "supportedFeatures": [
          "HTTPRouteMethodMatching"
        ],
        "unsupportedFeatures": [
          "HTTPRouteSchemeRedirect",
          "HTTPRoutePathRedirect",
          "HTTPRouteHostRewrite",
          "HTTPRoutePathRewrite",
          "HTTPRouteQueryParamMatching",
          "HTTPResponseHeaderModification",
          "HTTPRoutePortRedirect"
        ]
      }
```
would you move supported and unsupported features somewhere else?

Reply:
the current loc for `unsupportedFeatures` and `supportedFeatures` seems fine, Im having a hard time with `status.result` where `result` is the test output result which is `successful` is this case, but the `status` of the profile is `partial` (another synonym) since some features are unsupported

Reply:
sure would be good to get another pair of eyes on this PR to make sure my concern is a minor one that can be addressed later based on user feedback

Reply:
/cc @skriss

Reply:
It's explicitly part of the API definition that Extended features are _optional_, if we required you to have zero unsupported extended features to be marked as passing conformance, then that would break that contract.

To put this another way, it is absolutely not intended that _every_ implementation meet every feature, and meeting all the extended features will just mean that you'll have a bunch of ticks in the eventual implementation comparison matrix. I strongly disagree that "success" should be gated on supporting everything.